consensus/src/quorum_store/utils.rs

```diff
@@ -323,8 +328,23 @@ impl ProofQueue {
         ret
     }
 
+    pub(crate) fn num_total_txns(&mut self, current_time: LogicalTime) -> u64 {
+        let mut remaining_txns = 0;
+        for (digest, expiration) in self.digest_queue.iter() {
+            // Not expired
+            if *expiration >= current_time {
+                // Not committed
+                if let Some(Some(proof)) = self.digest_proof.get(digest) {
+                    remaining_txns += proof.info().num_txns;
+                }
+            }
+        }
```

Comment:
This may be fine for now, but we may need a TODO to optimize this, so we don't need to loop through the digest_queue to calculate the number of txns each time.